File: android/app/src/main/java/app/releaf/mobile/ui/components/NotebookRow.kt
/*
 * NotebookRow.kt
 *
 * List row for a notebook: icon chip + title + chapter tag + description
 * + meta + active status + chevron.
 *
 * Ported from Inkcreate mobile DS.
 */

package app.releaf.mobile.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Book
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppRadius
import app.releaf.mobile.ui.theme.AppSpacing
import app.releaf.mobile.ui.theme.AppTypography

@Composable
fun NotebookRow(
    title: String,
    meta: String,
    modifier: Modifier = Modifier,
    chapterTag: String? = null,
    description: String? = null,
    isActive: Boolean = false,
    icon: ImageVector = Icons.Filled.Book,
    onClick: () -> Unit = {},
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(AppRadius.md))
            .background(AppColors.CardSolid)
            .border(1.dp, AppColors.BorderDefault, RoundedCornerShape(AppRadius.md))
            .clickable(onClick = onClick)
            .padding(AppSpacing.s4),
        horizontalArrangement = Arrangement.spacedBy(AppSpacing.s3),
        verticalAlignment = Alignment.Top,
    ) {
        IconChip(icon = icon)

        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(AppSpacing.s1),
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(AppSpacing.s2),
            ) {
                Text(
                    text = title,
                    style = AppTypography.Button,
                    color = AppColors.TextPrimary,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                    modifier = Modifier.weight(1f, fill = false),
                )
                if (chapterTag != null) ChapterTag(chapterTag)
                Spacer(Modifier.weight(1f))
                if (isActive) ActivePill()
                Icon(
                    imageVector = Icons.Filled.ChevronRight,
                    contentDescription = null,
                    tint = AppColors.TextTertiary,
                    modifier = Modifier.size(16.dp),
                )
            }

            if (description != null) {
                Text(
                    text = description,
                    style = AppTypography.Meta,
                    color = AppColors.TextSecondary,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis,
                )
            }

            Text(
                text = meta,
                style = AppTypography.Meta,
                color = AppColors.TextTertiary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
    }
}

@Composable
private fun IconChip(icon: ImageVector) {
    Box(
        modifier = Modifier
            .size(40.dp)
            .clip(RoundedCornerShape(AppRadius.sm))
            .background(AppColors.CoralSoft),
        contentAlignment = Alignment.Center,
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = AppColors.CoralDeep,
            modifier = Modifier.size(18.dp),
        )
    }
}

@Composable
private fun ChapterTag(text: String) {
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(AppRadius.pill))
            .background(AppColors.CoralSoft)
            .padding(horizontal = AppSpacing.s2, vertical = 2.dp),
    ) {
        Text(text = text, style = AppTypography.Tag, color = AppColors.CoralDeep)
    }
}

@Composable
private fun ActivePill() {
    Row(
        modifier = Modifier
            .clip(RoundedCornerShape(AppRadius.pill))
            .background(AppColors.SuccessSoft)
            .padding(horizontal = AppSpacing.s2, vertical = 2.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Box(
            modifier = Modifier
                .size(6.dp)
                .clip(CircleShape)
                .background(AppColors.Success),
        )
        Text("Active", style = AppTypography.Tag, color = AppColors.Success)
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF5EEE3, widthDp = 390)
@Composable
private fun NotebookRowPreview() {
    Column(
        modifier = Modifier
            .background(AppColors.Canvas)
            .padding(AppSpacing.s4),
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s2),
    ) {
        NotebookRow(
            title = "Observability",
            chapterTag = "Ch. 1",
            description = "Dashboards, alerts, and on-call runbooks.",
            meta = "2 chapters · 3 pages · Updated 10 days ago",
            isActive = true,
        )
        NotebookRow(
            title = "Daily journal",
            meta = "1 chapter · 12 pages · Updated today",
        )
    }
}

File: apps/releaf/ios/Releaf/DesignSystem/Components/NotebookRow.swift
/*
 * NotebookRow.swift
 *
 * List row for a notebook.
 *
 * Layout:
 *   [icon chip] Title  [Ch. 1]           [Active] ›
 *               Description
 *               2 chapters · 3 pages · Updated 10 days ago
 *
 * Ported from Inkcreate mobile DS.
 */

import SwiftUI

public struct NotebookRow: View {
    public let title: String
    public let chapterTag: String?
    public let description: String?
    public let meta: String
    public let isActive: Bool
    public let systemIcon: String
    public let onTap: () -> Void

    public init(
        title: String,
        chapterTag: String? = nil,
        description: String? = nil,
        meta: String,
        isActive: Bool = false,
        systemIcon: String = "book.closed",
        onTap: @escaping () -> Void = {}
    ) {
        self.title = title
        self.chapterTag = chapterTag
        self.description = description
        self.meta = meta
        self.isActive = isActive
        self.systemIcon = systemIcon
        self.onTap = onTap
    }

    public var body: some View {
        Button(action: onTap) {
            HStack(alignment: .top, spacing: AppSpacing.s3) {
                iconChip

                VStack(alignment: .leading, spacing: AppSpacing.s1) {
                    HStack(spacing: AppSpacing.s2) {
                        Text(title)
                            .font(AppText.button)
                            .foregroundColor(AppColors.textPrimary)
                            .lineLimit(1)
                        if let chapterTag {
                            chapterTagPill(chapterTag)
                        }
                        Spacer(minLength: AppSpacing.s2)
                        if isActive { activePill }
                        Image(systemName: "chevron.right")
                            .font(.system(size: 13))
                            .foregroundColor(AppColors.textTertiary)
                    }

                    if let description {
                        Text(description)
                            .font(AppText.meta)
                            .foregroundColor(AppColors.textSecondary)
                            .lineLimit(2)
                    }

                    Text(meta)
                        .font(AppText.meta)
                        .foregroundColor(AppColors.textTertiary)
                        .lineLimit(1)
                }
            }
            .padding(AppSpacing.s4)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(
                RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                    .fill(AppColors.cardSolid)
            )
            .overlay(
                RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                    .stroke(AppColors.borderDefault, lineWidth: 1)
            )
            .appShadow(.sm)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }

    // MARK: Pieces

    private var iconChip: some View {
        ZStack {
            RoundedRectangle(cornerRadius: AppRadius.sm, style: .continuous)
                .fill(AppColors.coralSoft)
            Image(systemName: systemIcon)
                .font(.system(size: 18))
                .foregroundColor(AppColors.coralDeep)
        }
        .frame(width: 40, height: 40)
    }

    private func chapterTagPill(_ text: String) -> some View {
        Text(text)
            .font(AppText.tag)
            .foregroundColor(AppColors.coralDeep)
            .padding(.horizontal, AppSpacing.s2)
            .padding(.vertical, 2)
            .background(Capsule().fill(AppColors.coralSoft))
    }

    private var activePill: some View {
        HStack(spacing: 4) {
            Circle().fill(AppColors.success).frame(width: 6, height: 6)
            Text("Active")
                .font(AppText.tag)
                .foregroundColor(AppColors.success)
        }
        .padding(.horizontal, AppSpacing.s2)
        .padding(.vertical, 2)
        .background(Capsule().fill(AppColors.successSoft))
    }
}

#if DEBUG
struct NotebookRow_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: AppSpacing.s2) {
            NotebookRow(
                title: "Observability",
                chapterTag: "Ch. 1",
                description: "Dashboards, alerts, and on-call runbooks.",
                meta: "2 chapters · 3 pages · Updated 10 days ago",
                isActive: true
            )
            NotebookRow(
                title: "Daily journal",
                meta: "1 chapter · 12 pages · Updated today"
            )
        }
        .padding(AppSpacing.s4)
        .background(AppColors.canvas)
        .frame(width: 390)
    }
}
#endif
